tutorials/jmsActiveMQ/src/test/java/com/example/jmsActiveMQ/ApplicationTest.java

```diff
@@ -0,0 +1,205 @@
+      File file = new File(FILE_PATH);
+      file.delete();
+      
+      createAMQClient();
+      
+      // write messages to SQS Queue                                                                                                            
+      writeToQueue();
+
+      // run app asynchronously; terminate after results are checked                                                                              
+      LocalMode.Controller lc = asyncRun();
+
+      // check for presence of output file                                                                                                        
+      chkOutput();
+
+      // compare output lines to input                                                                                                            
+      compare();
+      
+      lc.shutdown();
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  /**
+   * Start JMS broker from here
+   *
+   * @throws Exception
+   */
+  private void startJMSService() throws Exception
+  {
+    broker = new BrokerService();
+    String brokerName = "ActiveMQOutputOperator-broker";
+    broker.setBrokerName(brokerName);
+    broker.getPersistenceAdapter().setDirectory(new File("target/activemq-data/" + 
+        broker.getBrokerName() + '/' + 
+        org.apache.activemq.store.kahadb.KahaDBPersistenceAdapter.class.getSimpleName()).getAbsoluteFile());
+    broker.addConnector("tcp://localhost:61617?broker.persistent=false");
+    broker.getSystemUsage().getStoreUsage().setLimit(1024 * 1024 * 1024);  // 1GB
+    broker.getSystemUsage().getTempUsage().setLimit(100 * 1024 * 1024);    // 100MB
+    broker.setDeleteAllMessagesOnStartup(true);
+    broker.start();
+  }
+  
+  private void createAMQClient() throws Exception 
+  {
+    startJMSService();
+    
+    // Create a ConnectionFactory
+    ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("vm://localhost");
+
+    // Create a Connection
+    connection = connectionFactory.createConnection();
+    connection.start();
+
+    // Create a Session
+    session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE);
+
+    // Create the destination (Topic or Queue)
+    Destination destination = session.createQueue(QUEUE_NAME);
+
+    // Create a MessageProducer from the Session to the Topic or Queue
+    producer = session.createProducer(destination);
+    producer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+  }
+
+  private void writeMsg(String[] msgs) throws JMSException 
+  {
+    for (String text : msgs) {
+      TextMessage message = session.createTextMessage(text);
+      producer.send(message);
+    }
+  }
+  
+  private void writeToQueue() throws JMSException {
+    writeMsg(lines);
+    
+    // Clean up
+    session.close();
+    connection.close();
```

Comment:
Considering the timing, it wouldn't make sense to move the clean up code to a separate @After function since the close needs to happen before the rest of the test code executes.